routes/cmip6.py

```diff
@@ -0,0 +1,187 @@
+                    di[model][scenario] = dict()
+
+                # Create an array of every month since January 1950 in the format "YYYY-MM"
+                months = [
+                    f"{year}-{str(month).zfill(2)}"
+                    for year in range(1950, 2100 + 1)
+                    for month in range(1, 13)
+                ]
+
+                for soi, value in enumerate(scenario_li):
+                    month = months[soi]
+                    if month not in di[model][scenario]:
+                        di[model][scenario][month] = dict()
+
+                    # The "ts" variable is still in Kelvin in the Rasdaman coverage.
+                    # Convert this to Celsius. All other temperature variables have
+                    # already been converted to Celsius before Rasdaman import.
+                    if varname == "ts" and float(value) != -9999:
+                        value = float(value) - 273.15
+
+                    # Evaporation has very tiny values.
+                    if varname == "evspsbl":
+                        precision = 7
+                    else:
+                        precision = 2
+
+                    di[model][scenario][month][varname] = round(float(value), precision)
+
+    return di
+
+
+@routes.route("/cmip6/point/<lat>/<lon>")
+def run_fetch_cmip6_monthly_point_data(lat, lon):
+    """
+    Query the CMIP6 monthly coverage
+
+    Args:
+        lat (float): latitude
+        lon (float): longitude
+
+    Returns:
+        JSON-like dict of requested CMIP6 monthly data
+
+    Notes:
+        example request: http://localhost:5000/cmip6/point/65.06/-146.16?vars=tas,pr
+    """
+    # Validate the lat/lon values
+    validation = validate_latlon(lat, lon)
+    if validation == 400:
+        return render_template("400/bad_request.html"), 400
+    if validation == 422:
+        return (
+            render_template(
+                "422/invalid_latlon.html", west_bbox=WEST_BBOX, east_bbox=EAST_BBOX
+            ),
+            422,
+        )
+    try:
+        var_parameter = False
+        if request.args.get("vars"):
+            var_parameter = True
+            vars = request.args.get("vars").split(",")
+            for var_id in vars:
+                if var_id not in varnames.values():
+                    return render_template("400/bad_request.html"), 400
+        else:
+            vars = None
+
+        if var_parameter:
+            results = {}
+            for var_id in vars:
+                var_coord = next(
+                    key for key, value in varnames.items() if value == var_id
+                )
+                point_data_list = asyncio.run(
+                    fetch_cmip6_monthly_point_data(lat, lon, var_coord)
+                )
+                new_results = package_cmip6_monthly_data(point_data_list, var_id)
+                for model, scenarios in new_results.items():
+                    results.setdefault(model, {})
```

Comment:
The `setdefault` method is new to me! That's cool. Seems like a handy way to dynamically create those nested dicts without having to mess with `collections.defaultdict`